python-main.js

```diff
@@ -461,18 +361,19 @@ function web_editor(config) {
                             setDescription(config.translate.drop.python);
                             reader.onload = function(e) {
                                 EDITOR.setCode(e.target.result);
-                            }
+                            };
                             reader.readAsText(f);
                             EDITOR.ACE.gotoLine(EDITOR.ACE.session.getLength());
                         } else if (ext == 'hex') {
                             setName(f.name.replace('.hex', ''));
                             setDescription(config.translate.drop.hex);
                             reader.onload = function(e) {
-                                var code = EDITOR.extractScript(e.target.result);
+                                var code = upyhex.extractPyStrFromIntelHex(
+                                        e.target.result);
                                 if(code.length < 8192) {
```

Comment:
Could we have a definition for the value `8192`? I assume this is the same value as `USER_CODE_LEN` from hexlify.js.
```javascript
var USER_CODE_LEN = 8 * 1024;
```
This is also used on line 549. 

How necessary is this? Shouldn't the error surface when the user tries to download the hex anyway? (see hexlify.js, line 105)

Reply:
Yes, we should remove magic numbers and get that value defined as a constant.
Because this area of the code will change when the new library is integrated I'll leave that to 
@microbit-sam to update.

@microbit-sam this type of issues would throw an error in the Filesystem library (https://github.com/microbit-foundation/js-microbit-micropython-fs/blob/master/src/appended-script.ts#L104), so when integrating it they should be captured and a warning shown to the user.

Reply:
Discussed with @microbit-sam, so marking as resolved.